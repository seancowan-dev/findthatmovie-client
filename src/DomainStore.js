import {observable,action,toJS} from 'mobx';
import uuid from "uuid";

class DomainStore{
    // App Primary 
     @observable updateType = "";
     @observable updateFolderId = "";
     @observable updateNoteId = "";
     @observable currentNote = null;
     @observable done = "";
     @observable updated = null;
     @observable errorMsg = "";
     @observable notes = {};
     @observable folders = {};

    // Update Component
     @observable folderNotes = [];
     @observable selectedFolder = "";
     @observable selectedFolderId = "";
     @observable selectedFolderIdOld = "";
     @observable oldNoteId = null;
     @observable currentNoteId = null;
     @observable migrateChecked = false;
     @observable note = {
         id: null,
         name: null,
         folderid: null,
         modified: null,
         content: null
     };

    // Local methods to use with fetch api methods
     handleErrors(response) { // prepares error message for HTTP request errors
      if (response.ok === true) {
          return response.json();
      } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
      }
    }
    hex(string) {
      let source = unescape(encodeURIComponent(string));
      let hex = '';
        for (var i = 0; i < source.length; i++) {
            hex += source.charCodeAt(i).toString(16)
        };
        return hex;
    }

  // Fetch API Methods

  @action async getFolders() {
    return await fetch(`http://localhost:8000/api/folders`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async getNotes() {
    return await fetch(`http://localhost:8000/api/notes`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async deleteNote(id) {
    return await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => { 
      if (response.ok === true) {
        this.promises();
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));

  }

  @action async addNote(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let date = new Date(Date.now());
    let note = {
      id: uuid.v4(),
      name: "",
      modified: date.toISOString(),
      folderid: "",
      content: ""
    };
    workingArr.filter(child => {
      if(child.name === "add-note-content") {
        note.content = child.value;
      }
      if (child.name === "add-note-title") {
        note.name = child.value;
      }
      if (child.name === "select-note-folder") {
        note.folderid = child.selectedOptions[0].id;
      }
      return null
    });
    await fetch(`http://localhost:8000/api/notes/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(note)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));    
  }

  @action async addFolder(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let folder = {
      id: "",
      name: "",
    };
    workingArr.filter(child => {
      if(child.name === "add-folder-title") {
        folder.name = child.value;
        folder.id = uuid.v4();
      }
      return null
    });

    await fetch(`http://localhost:8000/api/folders/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(folder)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));
  }

  @action async deleteFolder(id) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `DELETE`,
      headers: {
        'content-type': 'application/json'
      },

    })
    .then(response => { 
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateFolder(id, updates) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateNote(id, updates) {
    await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
        if (response.ok === true) {
          this.promises()
        } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
        }
    })
    .catch(e => alert(e));
  }

  @action promises() {
    Promise.all([this.getFolders(), this.getNotes()]).then(res => {
      this.folders = res[0];
      this.notes = res[1];
      this.done = true;
    });
  }
}
export default new DomainStore();