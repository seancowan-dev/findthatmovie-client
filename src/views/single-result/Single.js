import React from 'react';
import Search from '../../Search/Search';
import Result from '../../Results/Result/Result';
import Nav from '../../Nav/Nav';
import UserService from '../../services/users-service';
import TokenService from '../../services/token-service';
import { observer, inject } from 'mobx-react';

const Single = inject('userStore', 'searchStore', 'helpers')(observer((props) => {
        let moviePage
        let movieData
        let index;
        if (props.userStore.getUserInfo === null && props.userStore.getAuthenticated === true) { // If the user has not already visited the account page get their info
            UserService.getUserInfo(TokenService.readJwtToken().user_id)
            .then(res => {
                let userInfo = {
                    name: res.name,
                    email: res.email,
                    id: TokenService.readJwtToken().user_id,
                    created_at: res.created_at,
                    updated_at: res.updated_at,
                    perm_level: res.perm_level
                }
                props.userStore.setUserInfo(userInfo);
            })
        }
        
        if (props.searchStore.getSearchResults !== undefined) { // only map data if it is available
            moviePage = props.searchStore.getSearchResults.map((item, idx)=> {
                if (item.results.filter(item => {
                    if (item.original_title === unescape(props.title)) {
                        index = idx;
                        return item;
                    }
                    return null
                }) !== null) {
                    return item;
                }
                return null;
            });

        if (moviePage.length > 0 && props.searchStore.getLoading === true) {
                movieData = moviePage[index].results.filter(item => {
                    if (item.original_title === unescape(props.title)) {
                        return item;
                    }
                    return null
                });
                // Store the detailed info for Result component to use
                props.searchStore.setSingleMovieResults(movieData[0], movieData[0].id);
                props.searchStore.setLoading(false);
            }
        }

        

    return <>
        <Nav />
        <div className="parallax"></div>
        <div className="search-form-container"><Search /></div>
        <div className="results"><Result movie={props.title}/></div>
    </>
}));

export default Single;