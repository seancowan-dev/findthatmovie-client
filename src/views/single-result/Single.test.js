import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "mobx-react";
import DomainStore from '../../DomainStore';
import Single from './Single';

const store = {
    dataStore: DomainStore.dataStore,
    searchStore: DomainStore.searchStore,
    helpers: DomainStore.helpers,
    userStore: DomainStore.userStore,
    validators: DomainStore.validators
  }

describe('| Single Test Object |', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Provider {...store}><Single debug /></Provider>);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly in "production" mode', () => {
    const component = shallow(<Provider {...store}><Single production /></Provider>);
    expect(component).toMatchSnapshot();
  });
});