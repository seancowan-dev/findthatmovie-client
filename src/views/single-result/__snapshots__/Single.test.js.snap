// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`| Single Test Object | should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`| Single Test Object | should render correctly in "production" mode 1`] = `ShallowWrapper {}`;
